aten/src/ATen/TensorIndexing.h

```diff
@@ -219,7 +219,8 @@ static inline Tensor applySlice(
     SymInt length = (self_device == at::kCPU || self_device == at::kCUDA)
         ? (*self_sizes)[dim]
         : self.sym_size(dim);
-    if (!disable_slice_optimization && start == 0 && length == stop &&
+    if (!disable_slice_optimization &&
+        TORCH_GUARD_SIZE_OBLIVIOUS(start.sym_eq(0)) && length == stop &&
         step == 1) {
```

Comment:
curiosity q: I would naively have thought that we'll (eventually) need the same treatment for `step == 1` here. Is that wrong?

Reply:
Unclear. This could only matter if you had an unbacked SymInt specifying your step... which is a hard situation to imagine. (But yes, if it was unbacked, then making the guard here oblivious would be profitable.)